chocolate_distribution/chocolate_distribution.c

```diff
@@ -0,0 +1,116 @@
+#include <stdio.h>
+void merge_array(int a[], int l, int m, int r) 
+{ 
+    int i, j, k; 
+    int n1 = m - l + 1; 
+    int n2 =  r - m; 
+  
+    // array creation
+    int L[n1], R[n2]; 
+  
+    // Copy data to temp arrays L[] and R[] 
+    for (i = 0; i < n1; i++) 
+        L[i] = a[l + i]; 
+    for (j = 0; j < n2; j++) 
+        R[j] = a[m + 1+ j]; 
+  
+    /* Merge the temp arrays back into arr[l..r]*/
+    i = 0; 
+    j = 0;  
+    k = l; 
+    while (i < n1 && j < n2) 
+    { 
+        if (L[i] <= R[j]) 
+        { 
+            a[k] = L[i]; 
+            i++; 
+        } 
+        else
+        { 
+            a[k] = R[j]; 
+            j++; 
+        } 
+        k++; 
+    } 
+  
+    // Copy the remaining elements of L[], if there are any 
+    while (i < n1) 
+    { 
+        a[k] = L[i]; 
+        i++; 
+        k++; 
+    } 
+  
+    // Copy the remaining elements of R[], if there  are any 
+    while (j < n2) 
+    { 
+        a[k] = R[j]; 
+        j++; 
+        k++; 
+    } 
+} 
+  
+// l is for left index and r is right index of the sub-array of arr to be sorted 
+void merge_sort(int a[], int left, int right) 
+{ 
+    if (left < right) 
+    { 
+        // Same as (l+r)/2, but avoids overflow for 
+        // large l and h 
+        int m = left + (right-left) / 2; 
+  
+        // Sort first and second halves 
+        merge_sort(a, left, m); 
+        merge_sort(a, m+1, right); 
+  
+        merge_array(a, left, m, right); 
+    } 
+} 
+  
+int min_diff(int a[], int num_packets, int num_children)
+{
+    int i ,j ,diff = 0 ,required_diff ,first = 0 ,last = 0;

```

Comment:
In the previous review, I told you to add a space **after** every **comma**, not before them.